hazelcast/src/main/java/com/hazelcast/config/NearCacheConfig.java

```diff
@@ -159,7 +159,7 @@ public NearCacheConfig(int timeToLiveSeconds, int maxSize, String evictionPolicy
         if (evictionConfig != null) {
             this.evictionConfig = evictionConfig;
         } else {
-            this.evictionConfig.setSize(maxSize);
+            this.evictionConfig.setSize(calculateMaxSize(maxSize));
```

Comment:
This should convert the `0` from the deprecated config to `Integer.MAX_VALUE` for the new configuration via `EvictionConfig`, which doesn't allow `0` as size.